mtp_api/apps/performance/tests/test_views.py

```diff
@@ -166,3 +169,88 @@ def _monday_n_weeks_ago(self, weeks_ago):
 
         monday = monday_of_same_week(timezone.localdate())
         return monday - datetime.timedelta(weeks=weeks_ago)
+
+
+class UserSatisfactionUploadViewTestCase(AuthTestCaseMixin, APITestCase):
+
+    def setUp(self):
+        super().setUp()
+        create_super_admin()
+
+        self.upload_url = reverse('admin:user_satisfaction_upload')
+
+    def test_triggers_update_performance_data(self):
+        test_data = [
+            # +------------+----------+------------------+
+            # | start date | end date | expected week_to |
+            # +------------+----------+------------------+
+            ('2021-01-01', '2021-01-31', '2021-02-01'),  # last week is complete (Sunday)
+            ('2021-07-01', '2021-07-31', '2021-07-31'),  # last week is incomplete
+        ]
+        for (start_date, end_date, expected_week_to) in test_data:
+            test_csv = self.fake_csv(start_date, end_date)
+
+            self.client.login(username='admin', password='adminadmin')
+            self.client.post(
+                self.upload_url,
+                data={'csv_file': test_csv},
+            )
+
+            # Check 'update_performance_data' is scheduled correctly
+            job = ScheduledCommand.objects.get(name='update_performance_data')
+            self.assertEqual(job.arg_string, f'--week-from={start_date} --week-to={expected_week_to}')
+            self.assertTrue(job.delete_after_next)
+            self.assertEqual(job.cron_entry, '*/10 * * * *')
+            job.delete()
+
+    def fake_csv(self, start_date, end_date):
+        test_csv = StringIO()
+        test_csv.write('creation date,type,feedback\n')
+        test_csv.write(f'{start_date} 00:00:00,aggregated-service-feedback,Rating of 5: 99\n')
+        test_csv.write(f'{end_date} 00:00:00,aggregated-service-feedback,Rating of 5: 99\n')
+        test_csv.seek(0)
+
+        return test_csv
+
+
+class DigitalTakeupUploadViewTestCase(AuthTestCaseMixin, APITestCase):
+    fixtures = ['initial_types.json', 'test_nomis_mtp_prisons']
+
+    def setUp(self):
+        super().setUp()
+        create_super_admin()
+
+        self.upload_url = reverse('admin:digital_takeup_upload')
+        self.fixture_path = pathlib.Path(__file__).parent / 'files'
+        self.fixture_path = self.fixture_path / 'Money_to_Prisoner_Stats - 02.11.16.xls'
+
+        self.test_week = datetime.date(2016, 10, 31)  # Monday of week containing 2nd Nov 2016
+
+    def test_updates_performance_data_when_already_there(self):
+        PerformanceData.objects.create(week=self.test_week)
+
+        with open(self.fixture_path, 'rb') as f:
```

Comment:
fyi: you can open a `pathlib.Path` object directly with `self.fixture_path.open('rb')`. traditionally, you'd have `self.fixture_path` being a `str` and using the plain `open` builtin to open it. it amounts to the exact same effect, but technically this is a mix of traditional and newer styles.

Reply:
Thanks @ushkarev - admittedly I'm still somewhat a Python standard lib noob so these comments are gold 💯 

I've changed the test and amended that commit, so I'm afraid you'll have to have another look at the last commit/re-approve ☺️